README.md

```diff
@@ -6,22 +6,43 @@
-
-It's worth mentioning some of the tradeoffs w/r/t to reconciling virtual and
-actual state. Inputs are a bit wonky (but more on that in a later lesson, of
-course), and side effects in when rendering can grind an application to a halt.
-
-At the same time, JavaScript is fast while the DOM APIs are slow — this point is
-worth emphasizing.
+## What's a virtual DOM?
+![React VR?](https://media.giphy.com/media/3o7qDL7l1IZAQvxvj2/giphy.gif)
+
+By now, you should already know what the DOM is: a programmatic representation of the document we see in the browser.
+So, what's all this virtual DOM ruckus about? Well, it turns out that our good 'ol friend DOM is slow — terribly slow. He
+makes a granny using a walker look like Usain Bolt.
+
+Virtual DOM is a technique employed by several front-end libraries and frameworks, most notably React. In a nutshell,
+virtual DOM builds a _virtual_ representation of what our document should look like. When we're ready to render things
+to the screen, the virtual DOM will take a look at the existing DOM and change only what needs to be changed (in more
+technical terms, it's diffing and re-rendering the changes).
+
+The performance gains here are not to be underestimated, due to the way the virtual DOM works. Virtual DOM is created by
+a bunch of super smart people using extremely efficient algorithms. Most virtual DOMs also batch their updates to the
+DOM, ensuring that the _real_ DOM gets modified as little as possible. The point is: it's best to touch and change the
+DOM as little as possible, and when we do, we do it with surgical precision. Think of virtual DOM as our personal DOM
+surgeon!
+
+## So why use it?
+The above section should have already given you a good idea of when a virtual DOM is useful. It's a particularly great
+choice when we're building apps where changes in the interface happen a _lot_. By reducing the load on the real DOM, we
+can make our apps behave much more fluently and in line with what a user expects from software nowadays.
+
+In virtual DOM, we can just say 'okay, our document should look like _this_' — and the virtual DOM will do the heavy
+lifting for us (i.e. diffing the changes and applying them). This means that we don't need to write imperative code to
+update every tiny bit of our application, we just declare what the end result should look like, and the virtual DOM will
+do the rest!
+
+These abstractions also paved the road for non-DOM implementations for virtual DOM, for example React Native. In this
+case, it's easier to think of DOM as a generic document, rather than the thing we see in our web browser.
+
+## What's the catch?
+Even though virtual DOM is very performant and used in several high-profile libraries and frameworks, it's important to
+remember that it's still a _really_ clever workaround. There are other solutions in the works (like Shadow DOM) by
+browser vendors to make the DOM faster, but for right now, virtual DOM seems to be our best bet.
```

Comment:
Just a minor question, what's the difference between referring to "the Virtual DOM" and just "virtual DOM". Is one more correct than the other?